components/common-js/utils.js

```diff
@@ -6,4 +6,24 @@
  */
 const nodeList = nl => [].slice.call(nl || []);
 
-export { nodeList };
+/**
+ * throttle wrapper which we need for scrolling listeners: http://www.ianlopshire.com/javascript-scroll-events-doing-it-right/
+ *
+ * @param      {Function}  callback  The function to throttle
+ * @param      {Number}    wait      The minimum time that needs to pass between successive function calls
+ * @return     {Function}    { the throttled function }
+ */
+const throttle = (callback, wait) => {
+  let go = true;
+  return () => {
+    if (go) {
+      go = false;
+      setTimeout(() => {
+        go = true;
+        callback.call();
+      }, wait);
+    }
+  };
+};
```

Comment:
I don't get this one. Isn't `go` always `true`? Why l.19?

Reply:
it's a version of this https://remysharp.com/2010/07/21/throttling-function-calls

basically, `go` is true every time the `setTimeout` goes off, so that could be every 250ms. Once it goes off, the function is called, and `go` is set back to false, until the next 250ms pass. This prevents e.g. scroll events from firing too often and blocking the JS thread.